internal/controller/factory/statefulset.go

```diff
@@ -206,6 +153,115 @@ func generateEtcdArgs(cluster *etcdaenixiov1alpha1.EtcdCluster) []string {
 	return args
 }
 
+func generateContainers(cluster *etcdaenixiov1alpha1.EtcdCluster) []corev1.Container {
+	podEnv := []corev1.EnvVar{
+		{
+			Name: "POD_NAME",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					FieldPath: "metadata.name",
+				},
+			},
+		},
+		{
+			Name: "POD_NAMESPACE",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					FieldPath: "metadata.namespace",
+				},
+			},
+		},
+	}
+
+	containers := make([]corev1.Container, 0, len(cluster.Spec.PodTemplate.Spec.Containers))
+	for _, c := range cluster.Spec.PodTemplate.Spec.Containers {
+		if c.Name == "etcd" {
```

Comment:
Please consider using https://pkg.go.dev/k8s.io/apimachinery/pkg/util/strategicpatch for this